hooks/useCopyToClipBoard.ts

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<boolean>;
+
+function useCopyToClipBoard(): [CopiedValue, CopyFn] {
```

Comment:
This is awesome, thanks for creating a hook. I will replace my custom one on Homepage, Blog and Changelog click to copy thing.